tap_github/authenticator.py

```diff
@@ -41,7 +47,28 @@ def update_rate_limit(self, response_headers: Any) -> None:
         self.rate_limit_reset = int(response_headers["X-RateLimit-Reset"])
         self.rate_limit_used = int(response_headers["X-RateLimit-Used"])
 
-    def is_valid(self) -> bool:
+    def is_valid_token(self) -> bool:
```

Comment:
Some name swapping here:
- Renaming `is_valid` to `has_calls_remaining` because it's actually just checking the call budget.
- Naming this new function `is_valid_token` because it is actually testing tokens' validity.